crates/typst-cli/src/package.rs

```diff
@@ -53,13 +56,76 @@ impl PackageStorage {
 
         if let Some(cache_dir) = &self.package_cache_path {
             let dir = cache_dir.join(&subdir);
+            let parent_dir = dir.parent().unwrap();
+            let file_locking_err = |string| {
+                move |err| PackageError::FileLocking(eco_format!("{string}: {err}"))
+            };
+
+            let lock_file_path = parent_dir.join(format!(".{deepest_dir}.lock"));
+            let remove_lock = || {
+                fs::remove_file(&lock_file_path)
+                    .map_err(file_locking_err("failed to remove lock file"))
+            };
+            if !parent_dir.exists() {
+                fs::create_dir_all(parent_dir)
+                    .map_err(file_locking_err("failed to create parent directories"))?;
+            }
+            // https://github.com/yoshuawuyts/fd-lock/issues/28#issuecomment-2264180893
+            // File::create(&lock_file_path).unwrap();
+            // let file = match File::open(lock_file_path.as_path()) {
+            let file = fs::OpenOptions::new()
+                .create(true)
+                .truncate(true)
+                .write(true)
+                // .read(true) // to test the error
+                .open(lock_file_path.as_path())
+                .map_err(file_locking_err("failed to open lock file"))?;
+            let lock_file_path = format!(
+                "~{}",
+                lock_file_path.to_string_lossy().strip_prefix(env!("HOME")).unwrap()
+            );
+            // let mut lock_file = RwLock::new(file.try_clone().unwrap());
+            let mut lock_file = RwLock::new(file);
+            let try_lock = lock_file.try_write();
+            let _lock = match try_lock {
+                Ok(lock) => {
+                    println!("acquired the lock file {lock_file_path:?}");
```

Comment:
"Like that" means "at all" or you have a better idea? It's all just debugging, I don't know which messages will be left in the end.

Reply:
I see, I think in the end we definitely want to show that we are blocking when we wait for a lock file, but otherwise not at all. This is similar to how cargo does it.

Reply:
Yeah, I think so too, but I haven't thought about this yet.

Reply:
This is [marked as temporary](https://github.com/typst/typst/blob/7af1c0d938c4c5cdce047d54826656bef247be22/crates/typst-kit/src/package.rs#L131). We can discuss logging/messages when the core logic will 100% work.